### PR TITLE
feat(flashing): Flash RP2040 in Boot Mode

### PR DESCRIPTION
**Pull request by CODeRUS**

### CODE REVIEW

**Comment by dw-0** [comment]
Hi, thanks for the contribution!

I was thinking about, if you could maybe add an explanation in here:
https://github.com/dw-0/kiauh/blob/develop/kiauh/components/klipper_firmware/menus/klipper_flash_help_menu.py#L140

I think it would also be cool, if you could add one for the DFU connection, as if i remember correctly, you contributed that part to KIAUH too in the past :)

**Comment by CODeRUS** [comment]
@dw-0 added docs and rebased to develop

File: kiauh/components/klipper_firmware/menus/klipper_flash_help_menu.py [hunk 1]
```diff
@@ -143,6 +143,8 @@ def print_menu(self) -> None:
         count = 62 - len(color) - len(RESET_FORMAT)
         subheader1 = f"{COLOR_CYAN}USB:{RESET_FORMAT}"
         subheader2 = f"{COLOR_CYAN}UART:{RESET_FORMAT}"
+        subheader3 = f"{COLOR_CYAN}USB DFU:{RESET_FORMAT}"
+        subheader4 = f"{COLOR_CYAN}USB RP2 Boot:{RESET_FORMAT}"
```
Comment (dw-0) [inline kiauh/components/klipper_firmware/menus/klipper_flash_help_menu.py:147]: We should name it identically to the option that is selectable. So either rename the method to RP2 Boot or rename this heading to RP2040?

**Review by dw-0 — CHANGES_REQUESTED** [review]
Thanks for the quick update. I added one comment though.

**Comment by CODeRUS** [comment]
Renamed to RP2040, RP2 Boot is just a device name identifier in boot mode.

**Review by dw-0 — APPROVED** [review]

**Comment by dw-0** [comment]
Thank you!

**Comment by jrsall92** [comment]
Hi all,

seems like I can't update klipper on an RP2040 board (the one for positron if it's relevant) unless I double press the Reset button (which forces the board into bootloader). Is there something I'm doing wrong or is this the expected behaviour? On other boards like the Prusa Buddy and the FYSETC Cheetah 2 I can simply flash via Kiauh without any issues.

**Comment by CODeRUS** [comment]
@jrsall92 so what board you are talking about? Did you flashed it with zero boot offset? Also show last dmesg after executing normal flash.

**Pull request merged**